move BatchPermutationOp to caffe2/operators

wat3rBro opened the pull request:
Differential Revision: D19053527

kostmo commented:
## CircleCI build failures summary
As of commit 10c20cd:
* **1/1** failures introduced in this PR



## Detailed failure analysis

One may explore the probable reasons each build failed interactively [on the Dr. CI website](https://dr.pytorch.org/commit-details.html?sha1=10c20cd03e3661491bb5d6fd05ae60cff90ff82c).

### 1 failure *not* recognized by patterns:
| Job | Step | Status |
| --- | --- | --- |
| ![CircleCI](https://avatars0.githubusercontent.com/ml/7?s=12) <sup>pytorch_windows_test2</sup> | <sup>Test</sup> | New in PR |


---

<details><summary>This comment was automatically generated by <a href="https://github.com/kostmo/circleci-failure-tracker/tree/master/docs/from-pull-request-comment">Dr. CI</a> (expand for details).</summary>Follow <a href="https://dr.pytorch.org/admin/comments-opt-out.html">this link to opt-out</a> of these comments for your Pull Requests.

Please report bugs/suggestions on the <a href="https://github.com/kostmo/circleci-failure-tracker/issues">GitHub issue tracker</a>.
</details>

wat3rBro committed:
move BatchPermutationOp to caffe2/operators

Summary: Pull Request resolved: https://github.com/pytorch/pytorch/pull/31350

Differential Revision: D19053527

fbshipit-source-id: b62d9bf18b537de98b1da628d18ade800b0c460b

houseroad reviewed (approved):
LG
